executions/graphql-kotlin-transaction-batcher/src/main/kotlin/com/expediagroup/graphql/transactionbatcher/transaction/TransactionBatcher.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.transactionbatcher.transaction
+
+import com.expediagroup.graphql.transactionbatcher.publisher.TriggeredPublisher
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Type for [TransactionBatcher.queue] value, storing the [triggeredPublisher] instance
+ * and list of [transactions] that need to be executed by it
+ */
+data class TransactionBatcherQueueValue(
+    val triggeredPublisher: TriggeredPublisher<Any, Any>,
+    val transactions: MutableList<BatcheableTransaction<Any, Any>>
+)
+
+/**
+ * Holds logic to apply batching, deduplication and caching of [BatcheableTransaction]
+ * if no [TransactionBatcherCache] implementation is provided it will use [DefaultTransactionBatcherCache]
+ */
+class TransactionBatcher(
+    private val cache: TransactionBatcherCache = DefaultTransactionBatcherCache()
+) {
+
+    private val queue = ConcurrentHashMap<
```

Comment:
does order matter here?

Reply:
queue implies FIFO and it doesn't look like this variable is used as a queue, maybe we need a better name?

Reply:
> does order matter here?

Order in the queue (will rename it to batch) does not matter, what it matters is each publisher needs to produce data in the same order of inputs

Reply:
> queue implies FIFO and it doesn't look like this variable is used as a queue, maybe we need a better name?

i agree, probably just `batch` is a better name.